### PR TITLE
Add Rafter docs

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Unify Asset Store and Headless CMS docs into one Rafter component
- Rename DocTopic into AssetGroup
- Update labels
- Describe new AssetGroup functionality (adding BucketRef to AssetGroup CR definition)
- Review the existing docs and remove any outdated info
- Consolidate 4 tutorials for switching into Gateway mode into one with toggles for different cloud providers

TO DO (after Rafter is introduced in Kyma):
- Add an AssetGroup names `docs-components-rafter-cag.yaml` for Rafter under [`templates`](https://github.com/kyma-project/kyma/tree/master/resources/core/charts/docs/charts/content-ui/templates) with the **exact** content, asset order and sources as in [this](https://github.com/kyma-project/kyma/pull/6269/files#diff-b5a023f7f015635e8e7b2db2809f12a7) file (perhaps consider changing the order label into sth higher in the list, if available).
- Remove DocsTopics for Asset Store and HCMS from [`templates`](https://github.com/kyma-project/kyma/tree/master/resources/core/charts/docs/charts/content-ui/templates).

**Related issue(s)**
Relates to PR https://github.com/kyma-project/kyma/pull/6269 and https://github.com/kyma-project/kyma/pull/6418 with all post-review fixes.
See also #https://github.com/kyma-project/rafter/issues/10 & #5897

### CODE REVIEW

File: docs/rafter/05-01-rafter.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+| **controller-manager.envs.store.uploadWorkers** | Defines the number of workers used in parallel to upload files to the storage bucket. | `10` |
+| **controller-manager.envs.webhooks.validation.workers** | Defines the number of workers used in parallel to validate files. | `10` |
+| **controller-manager.envs.webhooks.mutation.workers** | Defines the number of workers used in parallel to mutate files. | `10` |
+| **upload-service.deployment.replicas** | Defines the number of service replicas. | `1` |
+| **front-matter-service.deployment.replicas** | Defines the number of service replicas. For more details, see the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/).| `1` |
+| **asyncapi-service.deployment.replicas** | Defines the number of service replicas. | `1` |
+| **asyncapi-service.envs.verbose** |  If set to `true`, you enable the extended logging mode that records more information on AsyncAPI Service activities than the usual logging mode which registers only errors and warnings. | `true` |
```
Comment (magicmatatjahu) [inline docs/rafter/05-01-rafter.md]: Please also add two additional parameters about verbose option in `front-matter` and `upload` services.

| Parameter | Description | Default value |
|-----------|-------------|---------------|
| **upload-service.envs.verbose** |  If set to `true`, you enable the extended logging mode that records more information on Upload Service activities than the usual logging mode which registers only errors and warnings. | `true` |
| **front-matter-service.envs.verbose** |  If set to `true`, you enable the extended logging mode that records more information on Front Matter Service activities than the usual logging mode which registers only errors and warnings. | `true` |

File: docs/rafter/05-01-rafter.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+| **controller-manager.envs.bucket.region.value** | Specifies the regional location of the bucket in a given cloud storage. Use one of [these](https://github.com/kyma-project/kyma/blob/master/resources/cluster-essentials/templates/bucket.crd.yaml#L52) regions. | `us-east-1` |
+| **controller-manager.envs.clusterBucket.maxConcurrentReconciles** | Defines the maximum number of cluster bucket concurrent reconciles which will run. | `1` |
+| **controller-manager.envs.bucket.maxConcurrentReconciles** | Defines the maximum number of bucket concurrent reconciles which will run. | `1` |
+| **controller-manager.envs.clusterAsset.maxConcurrentReconciles** | Defines the maximum number of cluster asset concurrent reconciles which will run. | `1` |
+| **controller-manager.envs.asset.maxConcurrentReconciles** | Defines the maximum number of asset concurrent reconciles which will run. | `1` |
+| **controller-manager.envs.store.secretKey** | Defines the secret key. Add the parameter to set your own **secretkey** credentials. | By default, **secretKey** is automatically generated. |
```
Comment (magicmatatjahu) [inline docs/rafter/05-01-rafter.md]: `secretKey` is not `controller-manager` inline but in `minio` chart of `controller-manager`, so `Parameter` is `controller-manager.minio.secretKey`

File: docs/rafter/05-01-rafter.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+| **controller-manager.envs.store.accessKey** | Defines the access key. Add the parameter to set your own **accesskey** credentials. | By default, **accessKey** is automatically generated. |
```
Comment (magicmatatjahu) [inline docs/rafter/05-01-rafter.md]: `accessKey` is not `controller-manager` inline but in `minio` chart of `controller-manager`, so `Parameter` is `controller-manager.minio.accessKey `

File: docs/rafter/08-01-add-new-documentation-to-console.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+         metadataWebhookService:
+         - endpoint: /v1/extract
+           filter: \.md$
+           name: rafter-front-matter-service
+           namespace: kyma-system
+         mode: package
+         url: https://github.com/prometheus/docs/archive/master.zip
+     status:
+       assetRef:
+         baseUrl: https://minio.kyma.local/rafter-public-1b7mtf1de5ost-1b7mtf1h187r7/prometheus-concepts-docs-markdown-1b7mu6bmkmse4
```
Comment (magicmatatjahu) [inline docs/rafter/08-01-add-new-documentation-to-console.md]: `https://storage.kyma.local`

File: docs/rafter/08-01-add-new-documentation-to-console.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+     resourceVersion: ""
+     selfLink: ""
+   ```
+
+   In the **status** section of the ClusterAsset, you can see details of all documents and **baseUrl** with their location in MinIO:
+
+   ```yaml
+   status:
+     assetRef:
+       baseUrl: https://minio.kyma.local/rafter-public-1b7mtf1de5ost-1b7mtf1h187r7/prometheus-concepts-docs-markdown-1b7mu6bmkmse4
```
Comment (magicmatatjahu) [inline docs/rafter/08-01-add-new-documentation-to-console.md]: `https://storage.kyma.local`

File: docs/rafter/08-01-add-new-documentation-to-console.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+status:
+  phase: Failed
+  reason: ValidationFailed
+  message: "The file is not valid against the provided json schema"
+```
+
+You can also analyze logs of the Rafter Controller Manager:
+
+```bash
+kubectl -n kyma-system logs -l 'app=rafter-controller-manager'
```
Comment (magicmatatjahu) [inline docs/rafter/08-01-add-new-documentation-to-console.md]: kubectl -n kyma-system logs -l 'app.kubernetes.io/name=rafter-controller-manager'

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 7]
```diff
@@ -0,0 +1,409 @@
+---
+title: Set MinIO to Gateway mode
+type: Tutorials
+---
+
+
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please remove blank line.

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 8]
```diff
@@ -0,0 +1,409 @@
...
+- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure)
+- [Microsoft Azure](http://azure.com) subscription
+
+  </details>
+  <details>
+  <summary label="aws-s3">
+  AWS S3
+  </summary>
+
+>**CAUTION:** AWS S3 Gateway mode was only tested manually on Kyma 1.6. Currently, there is no automated pipeline for it in Kyma.
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: AWS S3 Gateway mode was only tested manually on Kyma >= 1.6

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 9]
```diff
@@ -0,0 +1,409 @@
...
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [Amazon Web Services (AWS)](https://aws.amazon.com) account
+
+  </details>
+  <details>
+  <summary label="alibaba-cloud-oss">
+  Alibaba Cloud OSS
+  </summary>
+
+>**CAUTION:** Alibaba Cloud OSS Gateway mode was only tested manually on Kyma 1.6. Currently, there is no automated pipeline for it in Kyma.
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Alibaba Cloud OSS Gateway mode was only tested manually on Kyma >= 1.6

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 10]
```diff
@@ -0,0 +1,409 @@
...
+
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [Alibaba Cloud](https://alibabacloud.com) account
+
+  </details>
+
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Remove blank line

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 11]
```diff
@@ -0,0 +1,409 @@
...
+<div tabs name="create-required-cloud-resources" group="gateway-mode">
+  <details>
+  <summary label="google-cloud-storage">
+  Google Cloud Storage
+  </summary>
+
+Create a Google service account that has a private key and the **Storage Admin** role permissions. Follow these steps:
+
+1. Run the `export {VARIABLE}={value}` command to set up the following environment variables, where:
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please format whole list

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 12]
```diff
@@ -0,0 +1,409 @@
...
+
+  </details>
+  <details>
+  <summary label="azure-blob-storage">
+  Azure Blob Storage
+  </summary>
+
+Create an Azure resource group and a storage account. Follow these steps:
+
+1. Run the `export {VARIABLE}={value}` command to set up the following environment variables, where:
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please format whole list

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 13]
```diff
@@ -0,0 +1,409 @@
...
+metadata:
+  name: rafter-overrides
+  namespace: kyma-installer
+  labels:
+    installer: overrides
+    component: rafter
+    kyma-project.io/installation: ""
+type: Opaque
+data:
+  minio.gcsgateway.gcsKeyJson: "$GCS_KEY_JSON"
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please add `controller-manager.` prefix before all data fields.

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 14]
```diff
@@ -0,0 +1,409 @@
...
+kind: ConfigMap
+metadata:
+  name: rafter-overrides
+  namespace: kyma-installer
+  labels:
+    installer: overrides
+    component: rafter
+    kyma-project.io/installation: ""
+data:
+  minio.persistence.enabled: "false"
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please add `controller-manager.` prefix before all data fields.

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 15]
```diff
@@ -0,0 +1,409 @@
...
+metadata:
+  name: rafter-overrides
+  namespace: kyma-installer
+  labels:
+    installer: overrides
+    component: rafter
+    kyma-project.io/installation: ""
+type: Opaque
+data:
+  minio.accessKey: "$(echo "${AZ_ACCOUNT_NAME}" | base64)"
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please add `controller-manager.` prefix before all data fields.

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 16]
```diff
@@ -0,0 +1,409 @@
...
+kind: ConfigMap
+metadata:
+  name: rafter-overrides
+  namespace: kyma-installer
+  labels:
+    installer: overrides
+    component: rafter
+    kyma-project.io/installation: ""
+data:
+  minio.persistence.enabled: "false"
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please add `controller-manager.` prefix before all data fields.

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 17]
```diff
@@ -0,0 +1,409 @@
...
+metadata:
+  name: rafter-overrides
+  namespace: kyma-installer
+  labels:
+    installer: overrides
+    component: rafter
+    kyma-project.io/installation: ""
+type: Opaque
+data:
+  minio.accessKey: "$(echo "${AWS_ACCESS_KEY}" | base64)"
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please add `controller-manager.` prefix before all data fields.

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 18]
```diff
@@ -0,0 +1,409 @@
...
+kind: ConfigMap
+metadata:
+  name: rafter-overrides
+  namespace: kyma-installer
+  labels:
+    installer: overrides
+    component: rafter
+    kyma-project.io/installation: ""
+data:
+  minio.persistence.enabled: "false"
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please add `controller-manager.` prefix before all data fields.

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 19]
```diff
@@ -0,0 +1,409 @@
...
+metadata:
+  name: rafter-overrides
+  namespace: kyma-installer
+  labels:
+    installer: overrides
+    component: rafter
+    kyma-project.io/installation: ""
+type: Opaque
+data:
+  minio.accessKey: "$(echo "${ALIBABA_ACCESS_KEY}" | base64)"
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please add `controller-manager.` prefix before all data fields.

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 20]
```diff
@@ -0,0 +1,409 @@
...
+kind: ConfigMap
+metadata:
+  name: rafter-overrides
+  namespace: kyma-installer
+  labels:
+    installer: overrides
+    component: rafter
+    kyma-project.io/installation: ""
+data:
+  minio.persistence.enabled: "false"
```
Comment (magicmatatjahu) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]: Please add `controller-manager.` prefix before all data fields.

File: docs/rafter/03-03-assetgroup-cr-lifecycle.md [hunk 21]
```diff
@@ -0,0 +1,42 @@
+---
+title: AssetGroup custom resource lifecycle
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/03-03-assetgroup-cr-lifecycle.md]: as discussed, I will create a follow-up issue to adjust the `AssetGroup custom resource lifecycle` doc to the docs describing lifecycles for the Bucket and Asst CRs. I will then also move any additional info to a separate doc.

File: docs/rafter/03-04-minio-gateway.md [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+
+- To get the access key, run:
+  ```bash
+  kubectl get secret rafter-minio -n kyma-system -o jsonpath="{.data.accesskey}" | base64 -D
+  ```
+- To get the secret key, run:
+  ```bash
+  kubectl get secret rafter-minio -n kyma-system -o jsonpath="{.data.secretkey}" | base64 -D
+  ```
+
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/03-04-minio-gateway.md]: I created tabs, thx ;)

File: docs/rafter/03-04-minio-gateway.md [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+## Development mode storage
+
+MinIO is an open-source asset storage server with Amazon S3-compatible API. You can use it to store various types of assets, such as documents, files, or images.
+
+In the context of Rafter, the Asset Controller stores all assets in MinIO, in dedicated storage space.
+
+![](./assets/minio.svg)
+
+### Access MinIO credentials
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/03-04-minio-gateway.md]: As discussed, I rearranged the doc a bit :)

File: docs/rafter/08-02-set-minio-to-gateway-mode.md (hunk 8 above)
Comment (kazydek) [reply docs/rafter/08-02-set-minio-to-gateway-mode.md]: As discussed, we will keep it this way.

File: docs/rafter/08-02-set-minio-to-gateway-mode.md (hunk 9 above)
Comment (kazydek) [reply docs/rafter/08-02-set-minio-to-gateway-mode.md]: As discussed, we will keep it this way.

File: docs/rafter/06-01-asset.md [hunk 24]
```diff
@@ -0,0 +1,146 @@
...
+| **spec.source.metadatawebhookservice.filter** | No | Specifies the regex pattern used to select files sent to the service. |
+| **spec.bucketref.name** | Yes | Provides the name of the bucket for storing the asset. |
+| **status.phase** | Not applicable | The Asset Controller adds it to the Asset CR. It describes the status of processing the Asset CR by the Asset Controller. It can be `Ready`, `Failed`, or `Pending`. |
+| **status.reason** | Not applicable | Provides the reason why the Asset CR processing failed or is pending. See the [**Reasons**](#status-reasons) section for the full list of possible status reasons and their descriptions. |
+| **status.message** | Not applicable | Describes a human-readable message on the CR processing progress, success, or failure. |
+| **status.lastheartbeattime** | Not applicable | Provides the last time when the Asset Controller processed the Asset CR. |
+| **status.observedGeneration** | Not applicable | Specifies the most recent generation that the Asset Controller observes. |
+| **status.assetref** | Not applicable  | Provides details on the location of the assets stored in the bucket.   |
```
Comment (magicmatatjahu) [inline docs/rafter/06-01-asset.md]: **status.assetRef**

File: docs/rafter/06-01-asset.md [hunk 25]
```diff
@@ -0,0 +1,146 @@
...
+| **status.assetref.assets** | Not applicable | Provides the relative path to the given asset in the storage bucket. |
```
Comment (magicmatatjahu) [inline docs/rafter/06-01-asset.md]: **status.assetRef**

File: docs/rafter/06-01-asset.md [hunk 26]
```diff
@@ -0,0 +1,146 @@
...
+| **status.assetref.baseurl** | Not applicable | Specifies the absolute path to the location of the assets in the storage bucket. |
```
Comment (magicmatatjahu) [inline docs/rafter/06-01-asset.md]: **status.assetRef**

File: docs/rafter/06-04-clusterasset.md [hunk 27]
```diff
@@ -0,0 +1,143 @@
...
+| **spec.bucketref.name** | Yes | Provides the name of the bucket for storing the asset. |
+| **status.phase** | Not applicable | The ClusterAsset Controller adds it to the ClusterAsset CR. It describes the status of processing the ClusterAsset CR by the ClusterAsset Controller. It can be `Ready`, `Failed`, or `Pending`. |
+| **status.reason** | Not applicable | Provides the reason why the ClusterAsset CR processing failed or is pending. See the [**Reasons**](#status-reasons) section for the full list of possible status reasons and their descriptions.  |
+| **status.message** | Not applicable | Describes a human-readable message on the CR processing progress, success, or failure. |
+| **status.lastheartbeattime** | Not applicable | Provides the last time when the ClusterAsset Controller processed the ClusterAsset CR. |
+| **status.observedGeneration** | Not applicable | Specifies the most recent generation that the ClusterAsset Controller observes. |
+| **status.assetref** | Not applicable | Provides details on the location of the assets stored in the bucket.   |
```
Comment (magicmatatjahu) [inline docs/rafter/06-04-clusterasset.md]: **status.assetRef**

File: docs/rafter/06-04-clusterasset.md [hunk 28]
```diff
@@ -0,0 +1,143 @@
...
+| **status.assetref.assets** | Not applicable | Provides the relative path to the given asset in the storage bucket. |
```
Comment (magicmatatjahu) [inline docs/rafter/06-04-clusterasset.md]: **status.assetRef**

File: docs/rafter/06-04-clusterasset.md [hunk 29]
```diff
@@ -0,0 +1,143 @@
...
+| **status.assetref.baseurl** | Not applicable | Specifies the absolute path to the location of the assets in the storage bucket.   |
```
Comment (magicmatatjahu) [inline docs/rafter/06-04-clusterasset.md]: **status.assetRef**

**Pull request merged**